training-tracker-service/src/main/java/org/tolking/service/impl/TrainingEventServiceImpl.java

```diff
@@ -21,7 +22,7 @@
 @Service
 @RequiredArgsConstructor
 @Slf4j
-public class TrainingEventServiceImpl implements org.tolking.training_event_service.service.TrainingEventService {
+public class TrainingEventServiceImpl implements TrainingEventService {
```

Comment:
allright, you store each event in h2 and separate records. this is actually thread-safe.